### PR TITLE
Add more testing

### PR DESCRIPTION
**Pull request by Elkasitu**
lots a testing, need this merged so I can try out CI and coverage

### CODE REVIEW

File: misc.py [hunk 1]
```diff
@@ -89,6 +89,9 @@ class Colors:
         which require the color to be generated at runtime.
     """
 
+    def __init__(self):
+        raise NotImplementedError("Colors is a static class, it cannot be instantiated")
```
Comment (monzo94) [inline misc.py]: Surprised that this isn't the default when making an Enum class. What the hell python!

File: tests/test_entities.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    def test_actor_level_up_extra(self, player):
+        player.exp += 100
+        assert player.level == 2
+        assert player.exp == 17
+
+
+class TestBackpackItem(object):
+
+    def test_add_to_backpack(self, player, gold):
+        player.backpack.add(gold, 100)
```
Comment (monzo94) [inline tests/test_entities.py]: If we're gonna treat gold like an item in the backpack, maybe we should remove the gold attribute in Actor.

File: tests/test_entities.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import pytest
+
```
Comment (monzo94) [inline tests/test_entities.py]: Why the extra space? Thought you didn't like it.

File: tests/test_misc.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+
+import os
+import pytest
+
```
Comment (monzo94) [inline tests/test_misc.py]: [SPAAAAAAAAAAAAAACE](https://www.youtube.com/watch?v=4rmMQRs5qhI)

**Comment by monzo94** [comment]
Also, this closes #36

File: misc.py (hunk 1 above)
Comment (Elkasitu) [reply misc.py]: Actually it defaults to `__call__` which accepts Foo values and returns a Foo object containing that value, so I guess I'll remove this.

File: tests/test_entities.py (hunk 2 above)
Comment (Elkasitu) [reply tests/test_entities.py]: No I just used it as an example item, might as well call it `foo`

File: tests/test_entities.py (hunk 3 above)
Comment (Elkasitu) [reply tests/test_entities.py]: Only for `import foo` vs `from foo import bar`, but we can define the standard or just use PEP8 which is what you did originally I believe

**Comment by monzo94** [comment]
Also closes #4

**Comment by Elkasitu** [comment]
[CodeFactor](https://www.codefactor.io/repository/github/tonzo-studios/rogue-like) found an issue: Do not use `len(SEQUENCE)` to determine if a sequence is empty

It's currently on:
[tests\test_entities.py:124
](https://github.com/tonzo-studios/rogue-like/blob/ed72743d899ac872f058670b782dc87801e5e06a/tests/test_entities.py#L124)Commit ed72743d899ac872f058670b782dc87801e5e06a

File: backpack.py [hunk 5]
```diff
@@ -42,10 +46,9 @@ def add(cls, item_key, qty):
         if found and weight * qty + cls.cur_weight <= cls.max_weight:
             cls.contents[item_key] += qty
             cls.cur_weight += weight * qty
-        else:
-            if weight * qty + cls.cur_weight <= cls.max_weight:
-                cls.contents[item_key] = qty
-                cls.cur_weight += weight * qty
+        elif weight * qty + cls.cur_weight <= cls.max_weight:
+            cls.contents[item_key] = qty
+            cls.cur_weight += weight * qty
```
Comment (Elkasitu) [inline backpack.py]: We should log a message telling the player that his backpack is full

**Pull request merged**